### PR TITLE
Use curl progress callback directly to avoid cluttering the console on Windows, and drop threads

### PR DESCRIPTION
**Pull request by pzhlkj6612**
When using this program in CMD or PowerShell on Windows, its output sometimes becomes messy. Like this:

```console
PS > & .\Unlocker.exe --download-tools
auto-unlocker v1.1.3



Download progress: Download progress: Download progress: 2 %, 15.702 %, 16.292 %, 16.29 MB / 627.79 MB
Extracting from .tar to temp folder ...
Extracting from .zip to destination folder ...
Tools successfully downloaded!
Press enter to quit.
```

I tried `std::flush` , but it didn't solve the problem. Then I found that the original progress output of libcurl works fine on both Windows and Linux, so I realized that this may be a problem with multithreading. After simplifying the program, the output is no longer messy:

```console
PS > & .\Unlocker.exe --download-tools
auto-unlocker v1.1.3



Download progress: 5 %, 36.00 MB / 627.79 MB
```

```console
$ ./auto-unlocker --download-tools
auto-unlocker v1.1.3

The program is not running as root, the patch may not work properly.
Running the program with sudo/as root is recommended, in most cases required... Do you want to continue? (y/N) y


Download progress: 62 %, 390.68 MB / 627.79 MB
```

Since the beginning of this project ( https://github.com/paolo-projects/auto-unlocker/commit/811cd9f0e276453b877d941db329ecb0f0de503c#diff-9c936435ee95249f5cdc003b224042dd6b42660ffdc813d569d67758a890c633 ), you have been using another thread to output the progress. I'm curious about the reason for this, so it would be great if you could provide some explanation.

Since I removed the only part of this program which uses `<thread>` , I removed everything about threads as well.

We should close #50 if this PR is merged.

Configurations:

- auto-unlocker @ 4a94606e8b71662a92ea3cb445ae2923928d9ea8
- OS
  - Windows:
    - Windows 10 2004 10.0.19041.329
    - MSVC 19.26.28806.0
    - Windows SDK 10.0.18362.0
    - CMake 3.20.2
    - zlib 1.2.11 (source)
    - curl 7.74.0 & 7.76.1 (source)
    - libzip 1.7.3 (source)
  - Linux:
    - Kernel 5.8.0-50-generic
    - Ubuntu 20.04.0
    - gcc 9.3.0 (apt: gcc)
    - CMake 3.16.3 (apt: cmake)
    - zlib 1.2.11 (apt: zlib1g-dev)
    - curl 7.68.0 (apt: libcurl4-openssl-dev), 7.74.0 & 7.76.1 (source)
    - libzip 1.5.1 (apt: libzip-dev)

### CODE REVIEW

**Comment by paolo-projects** [comment]
The reason why I used an additional thread to update the progress is to avoid refreshing the output too frequently, which could be an issue, but I like the idea of removing the thread.
Writing to the console each time the callback is invoked results in the console being updated over 40k times per second!
I suggest at this point to remove the thread but only update the console when the MBs' count (taking into account the displayed precision) changes i.e. every 0.01+ MB downloaded

**Comment by pzhlkj6612** [comment]
Hi. Thank you for your reply!

> I suggest at this point to remove the thread but only update the console when the MBs' count (taking into account the displayed precision) changes i.e. every 0.01+ MB downloaded

I agree with you. I've experienced a high CPU usage on Windows due to the unlimited update frequency.  
Please see e4146d33a880f2a39b26befb382fbe40004b07c6 . I think updating progress per 100 kB might be reasonable.

**Comment by paolo-projects** [comment]
Yes now it seems good to me. The only minor issue is the cursor blinking at random positions due to a missing std::flush.
I could only try it on linux, did you test it on windows?

**Comment by pzhlkj6612** [comment]
Hi.

> The only minor issue is the cursor blinking at random positions due to a missing std::flush.

OK, I can reproduce it. It will be fixed right now.

> ... did you test it on windows?

Yes, but the cursor was invisible when downloading the tools.

File: src/netutils.cpp [hunk 1]
```diff
@@ -11,13 +11,25 @@ size_t Curl::write_data_file(char* ptr, size_t size, size_t nmemb, void* stream)
 	return bytes;
 }
 
+constexpr double mBytesProgressUpdateDelta = 0.1; // 100 kB
```
Comment (pzhlkj6612) [inline src/netutils.cpp:14]: Hmm, is 0.1 "MB" equal to 100 kB here? This seems a dumb question, but I just realized that the comment may be wrong.

**Comment by paolo-projects** [comment]
Well it's just 2 kbytes off, it won't do any harm for the purpose it is being used...

**Pull request merged**